### ISSUE TITLE
Select component problems

### ISSUE THREAD

**Issue by yannisbarlas** [Coko-Foundation/hhmi#317]
> **Migrated from GitLab**
> **Original Author:** Grigor Malo (`@grgml`)
> **Original Link:** https://gitlab.coko.foundation/coko-org/products/hhmi/hhmi/-/issues/164

As expected, the Select component was identified as problematic in the Deque accessibility assessment. I've contacted antd through their github repos to address the issues but response wasn't that positive. We got one issue fixed (which was also reported by previous accessibility assessments: the incorrect announcement of the number of options in the select dropdown), but they weren't willing to accept the other proposal. Their reasoning was that the said issue (missing and incorrect aria attributes due to a missing node) is detectable only from automatic tests, and that the node and the necessary attributes are added the moment the user interacts with the component, therefore assistive technologies would have no problems identifying the correct element. Which is true, but not ideal because it fails automatic tests, and they didn't give us an option to remedy the problem. 

The only proposal i have at this point is to grab the code from their repo and re-implement it with the necessary changes in our library. Or we can insist by opening other issues in their github about this problem until they take it seriously and do something about it.

FYI: `@DioneMentis`  `@yannis`

**Comment by yannisbarlas** [comment]
> **Migrated Comment**
> **Author:** aleezaoshry (`@aleezaoshry`)
> **Date:** 2023-01-17T06:16:46.430Z

`@DioneMentis`  `@grgml`  The audit included both manual and automated testing. I would suggest that we book consulting time with Deque to run through this and any other audit issues to explore more fully.

**Comment by yannisbarlas** [comment]
> **Migrated Comment**
> **Author:** Dione Mentis (`@DioneMentis`)
> **Date:** 2023-01-18T13:54:47.025Z

Sure `@aleezaoshry` , Can you please ask for their availability next week, US morning time. Thanks

**Comment by yannisbarlas** [comment]
> **Migrated Comment**
> **Author:** Grigor Malo (`@grgml`)
> **Date:** 2023-01-24T16:33:27.897Z

Hi `@aleezaoshry` , i will be at the meeting. Would it be ok if we set it for tomorrow, Wednesday, 1/25: 9 - 10 AM ET? Thanks!

**Comment by yannisbarlas** [comment]
> **Migrated Comment**
> **Author:** Grigor Malo (`@grgml`)
> **Date:** 2023-02-03T16:53:59.416Z

Thanks to the feedback from Deque the Select component doesn't throw any automatic accessibility errors anymore. Code has been merged to master and will be on staging as soon as the site is updated.

**Comment by yannisbarlas** [comment]
> **Migrated Comment**
> **Author:** Grigor Malo (`@grgml`)
> **Date:** 2023-05-30T10:31:44.250Z

Issues with the Select component were addressed in MR [#135](https://gitlab.coko.foundation/hhmi/hhmi/-/merge_requests/135) together with other accessibility improvements from the first Deque feedback. Closing the issue
